google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/Exceptions.java

```diff
@@ -94,23 +103,34 @@ protected SchemaMismatchedException(Status grpcStatus, String name) {
     }
   }
 
-  /** Offset already exists. */
+  /**
+   * Offset already exists. This happens when appending with offset that is less than the current
+   * end of the offset on the Stream. ou can also choose to append to the end of the Stream by not
+   * specifying offset.
+   */
   public static final class OffsetAlreadyExists extends StorageException {
     protected OffsetAlreadyExists(
         Status grpcStatus, String name, Long expectedOffset, Long actualOffset) {
       super(grpcStatus, name, expectedOffset, actualOffset, ImmutableMap.of());
     }
   }
 
-  /** Offset out of range. */
+  /**
+   * Offset out of range. This happens when appending with offset that is larger than the current
```

Comment:
Offset out of range.  This indicates that the append request is attempting to write data to a point beyond the current end of the stream.  To append data successfully, you must either specify the offset corresponding to the current end of stream, or omit the offset from the append request.